astropy/time/core.py

```diff
@@ -1269,13 +1310,13 @@ def __sub__(self, other):
         # we need a constant scale to calculate, which is guaranteed for
         # TimeDelta, but not for Time (which can be UTC)
         if other_is_delta:  # T - Tdelta
+            out = self.replicate()
             if self.scale in other.SCALES:
-                out = self.replicate()
                 if other.scale not in (out.scale, None):
                     other = getattr(other, out.scale)
             else:
-                out = getattr(self, (other.scale if other.scale is not None
-                                     else 'tai'))
+                out = getattr(out, (other.scale if other.scale is not None
```

Comment:
This was the subtle issue that referred to earlier, which took me a bit to track down.  When you do `out = getattr(self, 'tai')` here then `out` references the *same* object which gets cached.  Later on when `out._time.jd1` (and jd2) gets set, this of course updates the cached object, so the next time you do `getattr(self, 'tai')` you get the arithmetic result!